tests/test_io_sensors.py

```diff
@@ -0,0 +1,171 @@
+from .pigpio_mocks import *
+from vent.io.devices import ADS1015, ADS1115, PigpioConnection
+from vent.io.devices.sensors import Sensor, AnalogSensor, SFM3200, SimSensor
+
+
+def test_sensor():
+    """__________________________________________________________________________________________________________TEST #1
+    Tests errors are thrown when you try to do stuff with the abstract base class
+    """
+    with pytest.raises(TypeError):
+        s = Sensor()
+
+
+def test_sim_sensor(patch_pigpio_i2c, mock_i2c_hardware):
+    """__________________________________________________________________________________________________________TEST #2
+    Tests the proper functioning of a simulated sensor. Not much to this one, really. Also checks that age sends back -1
+        when update() has not been called
+    """
+    sim_sensor = SimSensor()
+    assert sim_sensor.age() == -1
+    n_iter = random.randint(1, 1000)
+    for i in range(n_iter):
+
+        result = sim_sensor.get()
+        assert sim_sensor.low <= result <= sim_sensor.high
```

Comment:
seems like this doesn't test against what the sensor should be reading, just that it's in range?

Reply:
Correct. SimSensor just returns a random value at the moment.